### PR TITLE
task.03.02.Halyna_Chervoniak.sql

### PR DESCRIPTION
**Pull request by Halynachervoniak**

### CODE REVIEW

File: Subtask.03.02.Halyna_Chervoniak.sql [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+FROM [Sales].[Customer] C
+    JOIN [Sales].[SalesOrderHeader] H
+    ON H.[CustomerID]=C.[CustomerID]
+    JOIN [Sales].[SalesPerson] SP
+    ON SP.[TerritoryID]=H.[TerritoryID]
+    JOIN [Person].[Person] P
+    ON  P.[BusinessEntityID]=SP.[BusinessEntityID]
+WHERE H.[OrderDate] BETWEEN '2014-01-01' AND '2014-12-31'
+ORDER BY [TotalDue] DESC
+OFFSET 3 ROWS FETCH NEXT 7 ROWS ONLY
```
Comment (makuzhel) [inline Subtask.03.02.Halyna_Chervoniak.sql:131]: перевір як ти обєднуєш таблиці, по яких ключах. шукаємо кастомерів з певною умовою.
результат має бути таким:
image

**Review by makuzhel — COMMENTED** [review]
reviewed

File: Subtask.03.02.Halyna_Chervoniak.sql [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+FROM [Production].[Product] P
+    JOIN [Sales].[SalesOrderDetail] OD
+      ON OD.[ProductID]=P.[ProductID]
+    JOIN  [Sales].[SalesOrderHeader] OH
+      ON OH.[SalesOrderID]=OD.[SalesOrderID]
+    JOIN [Sales].[PersonCreditCard] PC
+      ON PC.[CreditCardID]=OH.[CreditCardID]
+    JOIN [Sales].[CreditCard] C
+     ON C.[CreditCardID]=PC.[CreditCardID]
+WHERE [CardType]='Vista'
```
[the remark this answers is not in the document]
Comment (Halynachervoniak) [reply Subtask.03.02.Halyna_Chervoniak.sql:49]: SELECT DISTINCT P.[Name]
              , P.[Color]
FROM [Production].[Product] P
    JOIN [Production].[ProductSubcategory] PS ON PS.ProductSubcategoryID=p.ProductSubcategoryID AND PS.[ProductCategoryID] =1
    JOIN [Sales].[SalesOrderDetail] OD      ON OD.[ProductID]=P.[ProductID]
    JOIN  [Sales].[SalesOrderHeader] OH      ON OH.[SalesOrderID]=OD.[SalesOrderID]
    JOIN [Sales].[PersonCreditCard] PC      ON PC.[CreditCardID]=OH.[CreditCardID]
    JOIN [Sales].[CreditCard] C     ON C.[CreditCardID]=PC.[CreditCardID]
WHERE [CardType]='Vista'

File: Subtask.03.02.Halyna_Chervoniak.sql (hunk 1 above)
Comment (Halynachervoniak) [reply Subtask.03.02.Halyna_Chervoniak.sql:131]: SELECT   C.[CustomerID]
        --,[FirstName] + ' ' + ISNULL([MiddleName],' ') + ' ' + [LastName] AS [FullName] 
        ,H.[OrderDate]
        ,H.[TotalDue]
FROM [Sales].[SalesOrderHeader] H
JOIN [Sales].[Customer] C ON H.[CustomerID]=C.[CustomerID]
--JOIN [Person].[BusinessEntityContact] BE ON BE.PersonID=C.PersonID
--JOIN [Person].[Person] P    ON  P.BusinessEntityID=BE.BusinessEntityID
WHERE H.[OrderDate] BETWEEN '2014-01-01' AND '2014-12-31'
ORDER BY [TotalDue] DESC
OFFSET 3 ROWS FETCH NEXT 7 ROWS ONLY

File: Subtask.03.02.Halyna_Chervoniak.sql [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+HAVING COUNT([LastName])>1
+ORDER BY [LastName]
+
+SELECT [BusinessEntityID]
+      ,[FirstName] + ' ' + ISNULL([MiddleName],' ') + ' ' + [LastName] AS [FullName]
+FROM [Person].[Person] P
+WHERE [FirstName] IN ( SELECT        [FirstName]
+                       FROM [Person].[Person] P2
+                       WHERE [FirstName] like 'Z%')
+
```
[the remark this answers is not in the document]
Comment (Halynachervoniak) [reply Subtask.03.02.Halyna_Chervoniak.sql:79]: SELECT [ID]
     , [FullName]
     , P2.[BusinessEntityID]
     ,P2.[FirstName] + ' ' + ISNULL(P2.[MiddleName],' ') + ' ' + P2.[LastName] AS [FullName2]
FROM (
       SELECT [BusinessEntityID] AS [ID]
             ,[LastName] AS [LastName]
             ,[FirstName] + ' ' + ISNULL([MiddleName],' ') + ' ' + [LastName] AS [FullName]
            --, E.[JobTitle]
       FROM [Person].[Person] P
       ---JOIN [Person].[BusinessEntityContact] BE
       --ON BE.BusinessEntityID=P.BusinessEntityID
       WHERE [FirstName] like 'Z%'
) n
JOIN [Person].[Person] P2  ON P2.LastName=n.LastName AND P2.[BusinessEntityID]<>n.[ID]

**Pull request merged**